### DISCUSSION TITLE
CRON is back in the news

### DISCUSSION THREAD

**Discussion by f14m07** [liedekef/events-made-easy#331] [Q&A]
Hi,

Several months ago I had problems with eme_cron_send_queued being untimely deprogrammed.
I solved the problem with a snippet that reprogrammed the CRON. The snippet :

```php
add_action( 'eme_insert_rsvp_action', 'mon_test_cron' );
add_action('eme_insert_member_action','mon_test_cron' );
function mon_test_cron() {
   if (!wp_next_scheduled('eme_cron_send_queued')){
    wp_schedule_event(time(), 'eme_1min', 'eme_cron_send_queued');
    wp_mail('frederic.moriniere@gmail.com','CRON reprogrammé','eme_cron_send_queued reprogrammé');
}
}
```
After a few months of peace and quiet, I received the email again indicating that I could deprogram, but it doesn't seem to be working any more. I receive the email alerting me, but the CRON eme_cron_send_queued is not reprogrammed.

My CRON is triggered by the server every minute.

Here is an extract from my php error batch
```
[23-Oct-2023 08:31:44 UTC] Erreur d’événement de replanification de tâche pour le crochet eme_cron_send_queued. Code d’erreur : could_not_set. Message d’erreur : La liste des évènements planifiés n’a pas pu être sauvegardée.. Données : {"schedule":"eme_1min","args":[],"interval":60}
[23-Oct-2023 08:32:11 UTC] PHP Warning:  Undefined array key 0 in /home/taeq4768/vogue-et-reve.fr/wp-content/plugins/events-made-easy/eme-mailer.php on line 530
[23-Oct-2023 08:32:11 UTC] PHP Warning:  Trying to access array offset on value of type null in /home/taeq4768/vogue-et-reve.fr/wp-content/plugins/events-made-easy/eme-mailer.php on line 530

```
Any idea?

**Comment by liedekef** [comment]
Based on your first message, it is not only EME that had the reprogram issue:
```[23-Oct-2023 05:39:29 UTC] Erreur d’événement de replanification de tâche pour le crochet litespeed_task_imgoptm_req. Code d’erreur : could_not_set. Message d’erreur : La liste des évènements planifiés n’a pas pu être sauvegardée.. Données : {"schedule":"litespeed_filter","args":[],"interval":60}```

So, it's fairly safe to say it is not an EME issue, but a WP issue. WP is known for its flawed cron (it is 1 entry in the settings, causing issues to pop up all the time ... it baffles me that they don't fix this). Make sure not to plan things too fast, I'm guessing systems with lots of crons will have issues (might even be a caching issue if you have some DB caching plugin).
i used to have similar issues and tried and tried until I removed some double cron entries and disabled plugins that caused too much cron entries to run too fast.

**Reply by liedekef** [reply]
See also: https://core.trac.wordpress.org/ticket/57271 
In fact, nobody seems to realise that the way it is set up, wp will always have cron issues ...

**Reply by f14m07** [reply]
It's quite true that it's not the only function to have this problem. The other is the Lite Speed cache plugin. What I'd like to understand is why my snippet isn't programming eme_cron_send_queued again even though it's sending me the email. Have I programmed it wrong?

**Reply by liedekef** [reply]
It seems just fine to me, I use the same logic.
Most people tend to think it is a db-locking issue, I agree on the part it is most probably db-related (but in the end a wp issue).
Currently, go in the EME settings and set the cron queue time from there, maybe that works?

**Reply by f14m07** [reply]
Yes, that's what I do when I realise that the queue programming has been lost. But I thought that the 
```
wp_schedule_event(time(), 'eme_1min', 'eme_cron_send_queued');
```
function was enough. 
Maybe I should use a different hook?

**Reply by liedekef** [reply]
It might be possible that the hook is too late at that moment ... the doc is very unclear there. I know it works ok during plugin activation/deactivation and settings save, but otherwise I've struggled with this too ...
